pyiceberg/io/pyarrow.py

```diff
@@ -387,15 +387,30 @@ def __init__(self, properties: Properties = EMPTY_DICT):
         super().__init__(properties=properties)
 
     @staticmethod
-    def parse_location(location: str) -> Tuple[str, str, str]:
-        """Return the path without the scheme."""
+    def parse_location(location: str, properties: Properties = EMPTY_DICT) -> Tuple[str, str, str]:
+        """Return (scheme, netloc, path) for the given location.
+
+        Uses environment variables DEFAULT_SCHEME and DEFAULT_NETLOC
+        if scheme/netloc are missing.
+        """
         uri = urlparse(location)
-        if not uri.scheme:
-            return "file", uri.netloc, os.path.abspath(location)
-        elif uri.scheme in ("hdfs", "viewfs"):
-            return uri.scheme, uri.netloc, uri.path
+
+        # Load defaults from environment
+        default_scheme = properties.get("DEFAULT_SCHEME", "file")
+        default_netloc = properties.get("DEFAULT_NETLOC", "")
+
+        # Apply logic
+        scheme = uri.scheme or default_scheme
+        netloc = uri.netloc or default_netloc
```

Comment:
wdyt about keeping this logic inside of an
```
if not uri.scheme:
    scheme = properties.get("DEFAULT_SCHEME", "file") 
    netloc = uri.properties.get("DEFAULT_NETLOC", "")
```
block? 

since this is only applicable in the case where path without scheme is ambiguous. 
the logic is never applied when the scheme is present

Reply:
fixed